src/handlers/gestures/simpleGestures.ts

```diff
@@ -0,0 +1,429 @@
+  onStart?: (event: HandlerStateChangeEventPayload) => void;
+  onEnd?: (event: HandlerStateChangeEventPayload, success: boolean) => void;
+  onUpdate?:
+    | ((event: PanGestureHandlerEvent) => void)
+    | ((event: PinchGestureHandlerEvent) => void)
+    | ((event: RotationGestureHandlerEvent) => void)
+    | ((event: ForceTouchGestureHandlerEvent) => void);
+};
+
+export abstract class SimpleGesture extends Gesture {
+  public handlerTag = -1;
+  public handlerName = '';
+  public config: Partial<CommonGestureConfig> = {};
+  public handlers: HandlerCallbacks = {
+    handlerTag: -1,
+  };
+
+  private addDependency(
+    key: 'simultaneousWith' | 'requireToFail',
+    gesture: SimpleGesture | React.RefObject<SimpleGesture>
+  ) {
+    this.config[key] = this.config[key]
+      ? [].concat(this.config[key], gesture)
+      : [gesture];
+  }
+
+  private toArray(x: unknown) {
+    return [].concat(x);
+  }
+
+  setRef(ref: React.RefObject<SimpleGesture>) {
+    this.config.ref = ref;
+    return this;
+  }
+
+  setOnBegan(callback: (event: HandlerStateChangeEventPayload) => void) {
+    this.handlers.onBegan = callback;
+    return this;
+  }
+
+  setOnStart(callback: (event: HandlerStateChangeEventPayload) => void) {
+    this.handlers.onStart = callback;
+    return this;
+  }
+
+  setOnEnd(
+    callback: (event: HandlerStateChangeEventPayload, success: boolean) => void
+  ) {
+    this.handlers.onEnd = callback;
+    return this;
+  }
+
+  setEnabled(enabled: boolean) {
+    this.config.enabled = enabled;
+    return this;
+  }
+
+  setMinPointers(minPointers: number) {
+    this.config.minPointers = minPointers;
+    return this;
+  }
+
+  setShouldCancelWhenOutside(value: boolean) {
+    this.config.shouldCancelWhenOutside = value;
+    return this;
+  }
+
+  setHitSlop(hitSlop: HitSlopType) {
+    this.config.hitSlop = hitSlop;
+    return this;
+  }
+
+  addSimultaneousGesture(
+    gesture: SimpleGesture | React.RefObject<SimpleGesture>
+  ) {
+    this.addDependency('simultaneousWith', gesture);
+
+    return this;
+  }
+
+  addRequiredToFailGesture(
+    gesture: SimpleGesture | React.RefObject<SimpleGesture>
+  ) {
+    this.addDependency('requireToFail', gesture);
+
+    return this;
+  }
+
+  simultaneousWith(other: SimpleGesture): GestureBuilder {
+    return new GestureBuilder(this).simultaneousWith(other);
+  }
+
+  exclusiveWith(other: SimpleGesture): GestureBuilder {
+    return new GestureBuilder(this).exclusiveWith(other);
+  }
+
+  requireToFail(other: SimpleGesture): GestureBuilder {
+    return new GestureBuilder(this).requireToFail(other);
+  }
+
+  initialize() {
+    this.handlerTag = getNextHandlerTag();
+    this.handlers = { ...this.handlers, handlerTag: this.handlerTag };
+
+    if (this.config.ref) {
+      this.config.ref.current = this;
+    }
+  }
+
+  build(): BuiltGesture {
+    const result = new BuiltGesture(this.prepare);
+    result.gestures = [this];
+    return result;
```

Comment:
```suggestion
    const gestureConfig = new BuiltGesture(this.prepare);
    gestureConfig.gestures = [this];
    return gestureConfig;
```

Reply:
Fixed in 179e9e4d8e7b6646b137064bd7afc332629d2129